BrowserStack.java

```diff
@@ -34,36 +34,83 @@ public static void myassert(boolean  x) {
             throw new IllegalArgumentException("Assert fail") ;
         }
     }
+    
     //push() - pushes the element in this case its a url (String)
     private  void push(String url){
-        //WRITE YOUR CODE HERE
+        top++;
+        myStack[top] = url;
     }
+
     //pop()- pops the element from the stack
     private String pop(){
-        //WRITE YOUR CODE HERE
+        String popElement = myStack[top];
+        top--;
+        return popElement;
 
     }
-
+    
+    //print stack elements
     private void printStack(){
-        //WRITE YOUR CODE HERE
-
+        int i;
+        System.out.println("Stack Elements: ");
+        for(i = top; i >=0; i--){
+            System.out.println(myStack[i]);
+        }
     }
-    private  String peek(){
-        //WRITE YOUR CODE HERE
 
+    //peek at the top element
+    private  String peek(){
+        System.out.println("Peek: "+myStack[top]);
+        System.out.println("After peek: ");
+        printStack();
+        return myStack[top];
     }
+    
+    //size of the stack
     private  int size(){
-        //WRITE YOUR CODE HERE
-
+        return top+1;
     }
+    
+    //check whether the stack is Empty
     private  boolean isEmpty(){
-        //WRITE YOUR CODE HERE
-
+        int stackSize = size();
+        boolean empty = false;
+        if(stackSize == 0){
+            System.out.println("Empty Stack");
+            empty = true;
+        }
+        else{
+            System.out.println("Stack is not empty.");
+        }
+        return empty;
     }
-    private  boolean search(String url){
-        //WRITE YOUR CODE HERE
 
+    //Search an element 
+    private boolean search(String url){
+        int i = 0, j = 0;
+        String[] tempStack = new String[16];
```

Comment:
If you are using a new stack to search, then it sufficient to initialze with the size of 'top+1'.